synapse/lib/types.py

```diff
@@ -80,38 +80,90 @@ def __init__(self, tlib, name, **info):
         self._getTufoByProp = getattr(tlib, 'getTufoByProp', None)
+        self._getSubProps = getattr(tlib, 'getSubProps', None)
+        self._getPropNorm = getattr(tlib, 'getPropNorm', None)
 
     def norm(self, valu, oldval=None):
 
+        if isinstance(valu, (list, tuple)):
+            return self._norm_list(valu, oldval)
+
         if not isinstance(valu, str) or len(valu) < 1:
             self._raiseBadValu(valu)
 
+        return self._norm_str(valu, oldval)
+
+    def _norm_str(self, text, oldval=None):
+        text = text.strip()
+        if not text:
+            self._raiseBadValu(text, mesg='No text left after strip().')
+
+        if text[0] == '(':
+            vals, off = s_syntax.parse_list(text)
+            if off != len(text):
+                self._raiseBadValu(text, off=off, vals=vals,
+                                   mesg='List parting for guid type did not consume all of the input text.')
+            return self._norm_list(vals, oldval)
+
         # generate me one.  we dont care.
-        if valu == '*':
+        if text == '*':
             return s_common.guid(), {}
 
-        if valu[0] != '$':
-            retn = valu.lower().replace('-', '')
+        if text[0] != '$':
+            retn = text.lower().replace('-', '')
             if not isguid(retn):
-                self._raiseBadValu(valu)
+                self._raiseBadValu(text)
 
             return retn, {}
 
         if self._guid_alias is None:
-            self._raiseBadValu(valu, mesg='guid resolver syntax used with non-aliased guid')
+            self._raiseBadValu(text, mesg='guid resolver syntax used with non-aliased guid')
 
         if self._getTufoByProp is None:
-            self._raiseBadValu(valu, mesg='guid resolver syntax used with non-cortex tlib')
+            self._raiseBadValu(text, mesg='guid resolver syntax used with non-cortex tlib')
 
         # ( sigh... eventually everything will be a cortex... )
-        node = self._getTufoByProp(self._guid_alias, valu[1:])
+        node = self._getTufoByProp(self._guid_alias, text[1:])
         if node is None:
-            self._raiseBadValu(valu, mesg='no result for guid resolver',
+            self._raiseBadValu(text, mesg='no result for guid resolver',
                                alias=self._guid_alias)
 
         iden = node[1].get(node[1].get('tufo:form'))
         return iden, {}
 
+    def _norm_list(self, valu, oldval=None):
+
+        if not valu:
+            self._raiseBadValu(valu=valu, mesg='No valus present in list to make a guid with')
+
+        subprops = self._getSubProps(self.name)
+        if not subprops:
+            self._raiseBadValu(valu,
+                               form=self.name,
+                               mesg='Cannot norm a guid valu as a list if there are no subprops for the form'
+                               )
+        valid_props = set()
+        for fullprop, pnfo in subprops:
+            valid_props.add(pnfo.get('relname'))
```

Comment:
This uses the `rel:prop=valu` syntax, instead of `:rel:prop=valu` syntax.  Do we want to enforce the second one here?

Reply:
@therealsilence Do you have any input on this?

Reply:
No strong feelings...if by "stable guid generation" we're (essentially, from a dumb analyst perspective) saying if some set of "foothings" are input as elements to a comp type (regular or comp with optionals), then the guid should be predictable based on the input" then I agree with @invisig0th that it's helpful to keep the same syntax (no colon) as "similar" processes.

Reply:
okie - will leave these as-is and avoid the `:rel:prop` syntax.